include/znc/Modules.h

```diff
@@ -603,6 +603,16 @@ class CModule {
                          CChan& Channel, bool bNoChange);
     virtual void OnDeop(const CNick& OpNick, const CNick& Nick, CChan& Channel,
                         bool bNoChange);
+    /** Called when a nick is halfopped on a channel */
+    virtual void OnHalfOp2(const CNick* pOpNick, const CNick& Nick, CChan& Channel,
+                       bool bNoChange);
+    virtual void OnHalfOp(const CNick& OpNick, const CNick& Nick, CChan& Channel,
+                      bool bNoChange);
+    /** Called when a nick is dehalfopped on a channel */
+    virtual void OnDeHalfOp2(const CNick* pOpNick, const CNick& Nick,
+                         CChan& Channel, bool bNoChange);
+    virtual void OnDeHalfOp(const CNick& OpNick, const CNick& Nick, CChan& Channel,
```

Comment:
If adding new callbacks, need to update modperl and modpython

Reply:
modperl and modpython updated

Reply:
You've missed the `function.in` files.